### PR TITLE
chore(vdev): Regularize vdev integration test arguments to cargo

### LINKED ISSUE #16147: Regularize `vdev` integration test arguments to `cargo`
In the `test.yaml` file for the integration tests, every copy has an `args` list containing the same pattern:
* `--features FLAGS`
* Either `--lib` or `--test NAME`
* A test name filter (ex `::elasticsearch::`)

These should get refactored out into custom fields instead of a generic `args` block:
* A list of features
* An optional test name, if not present runs the lib tests
* A test filter string

These would be turned into the same parameters to the `cargo` command line, but under program control.

### PR DESCRIPTION
**Pull request by jonathanpv**
Fixes the prometheus integration breakage introduced earlier from https://github.com/vectordotdev/vector/pull/16570
by appending test_filter to the cargo commands. 

- Added fields to the struct `IntegrationTestConfig`
- Added logic to handle case if the `--test` field was not present, defaults to `--lib`

This is a draft PR at the moment to showcase that the code works for two test integrations namely shutdown and redis, before I make the changes on the rest of the integrations config files.

## Testing
There's two config files I modified for testing purposes. Namely `scripts/integration/redis/test.yaml` and `scripts/integration/shutdown/test.yaml` since these cover the cases of using a test_filter pattern, the `--lib` flag, and the `--test NAME` flag.

Before I was able to test I need to start a docker daemon
```
 docker run -d -p 80:80 docker/getting-started
```

Then I ran
```
cargo vdev integration start redis
```

Then for the actual testing we must run
```
cargo vdev int test redis
```

Closes #16147

### CODE REVIEW

File: vdev/src/testing/integration.rs [hunk 1]
```diff
@@ -64,7 +64,23 @@ impl IntegrationTest {
...
+        match self.config.test {
+            Some(ref test_arg) => {
+                args.push("--test".to_string());
+                args.push(test_arg.to_string());
+            }
+            None => args.push("--lib".to_string()),
+        }
+
+        // Ensure the test_filter args are passed as well
+        args.push(self.config.test_filter.clone().unwrap_or("".to_string()));
```
Comment (bruceg) [inline vdev/src/testing/integration.rs:83]: If `test_filter` is `None`, we want to skip pushing the arg, not push an empty string.

File: vdev/src/testing/integration.rs [hunk 2]
```diff
@@ -64,7 +64,23 @@ impl IntegrationTest {
         for (key, value) in config_env(&self.env_config) {
             env_vars.insert(key, Some(value));
         }
-        let mut args = self.config.args.clone();
+
+        let mut args = self.config.args.clone().unwrap_or_default();
+
+        args.push("--features".to_string());
```
Comment (bruceg) [inline vdev/src/testing/integration.rs:70]: AFAICR `args.push` supports pushing static strings too, so I think this (and below) could just be:
```suggestion
        args.push("--features");
```
IOW I don't think the `.to_string()` conversions are strictly necessary for things that are already strings.

**Review by bruceg — CHANGES_REQUESTED** [review]
Looks good except for a couple of cleanups in `integration.rs`

File: vdev/src/testing/integration.rs (hunk 2 above)
Comment (jonathanpv) [reply vdev/src/testing/integration.rs:70]: I don't think it's possible as I tried this and was met with the following message from the compiler:

error[E0308]: mismatched types
    --> vdev/src/testing/integration.rs:70:19
     |
     |         args.push("--features");
     |              ---- ^^^^^^^^^^^^- help: try using a conversion method: `.to_string()`
     |              |    |
     |              |    expected struct `std::string::String`, found `&str`
     |              arguments to this function are incorrect

Comment (bruceg) [reply vdev/src/testing/integration.rs:70]: Oh, *sigh* I was thinking this was `std::process::Command` which can add either kind of arg, but what you have here is `Vec<String>.push` Of course you are right.

**Review by bruceg — APPROVED** [review]

**Comment by jonathanpv** [comment]
🎉  I'm so happy this is finally merged in, I'll keep the momentum high and try to merge as many PRs as possible 👨‍💻

**Pull request merged**